dymos/visualization/timeseries_plots.py

```diff
@@ -0,0 +1,193 @@
+                if phases_node:
+                    return phases_node, phase_node_path
+    return None, None
+
+
+def timeseries_plots(solution_recorder_filename, simulation_record_file=None, plot_dir="plots"):
+    """
+    Given timeseries data from case recorder files, make separate plots of each variable
+    and store the plot files in the directory indicated by the variable plot_dir
+
+    Parameters
+    ----------
+    solution_recorder_filename : str
+        The path to the case recorder file containing solution data
+    simulation_record_file : str or None (default:None)
+        The path to the case recorder file containing simulation data. If not None,
+        this implies that the data from it should be plotted
+    plot_dir : str
+        The path to the directory to which the plot files will be written
+    """
+
+    # get ready to generate plot files
+    plot_dir_path = os.path.join(os.getcwd(), plot_dir)
+    if not os.path.isdir(plot_dir_path):
+        os.mkdir(plot_dir_path)
+
+    cr = om.CaseReader(solution_recorder_filename)
+
+    # get outputs from the solution
+    solution_cases = cr.list_cases('problem')
+    last_solution_case = cr.get_case(solution_cases[-1])
+
+    # If plotting simulation results, get the values for those variables
+    if simulation_record_file:
+        cr_simulate = om.CaseReader(simulation_record_file)
+        system_simulation_cases = cr_simulate.list_cases('problem')
+        last_simulation_case = cr_simulate.get_case(system_simulation_cases[-1])
+
+    # we will use the problem metadata to get information about the phases and units
+    root_children = cr.problem_metadata['tree']['children']
+
+    # We have to key off the phases node. It will tell us
+    #  what the prefix is for the variable names before the phases part of the variables
+    #  and also let us know what the phase names are
+    phases_node, phases_node_path = _get_phases_node_in_problem_metadata(root_children)
+
+    # get phase names
+    phase_names = [phase['name'] for phase in phases_node['children']]
+
+    # Get the units and var names along the way
+    var_units = {}
+    time_units = {}
+    for phase_node in phases_node['children']:  # Hopefully all phases have the same vars
+        for phase_node_child in phase_node['children']:  # find the timeseries node
+            if phase_node_child['name'] == 'timeseries':
+                timeseries_node = phase_node_child
+                break
+        # get the time units first so they can be associated with all the variables
+        for timeseries_node_child in timeseries_node['children']:
+            if timeseries_node_child['name'] == 'time':
+                units_for_time = timeseries_node_child['units']
+                break
+        for timeseries_node_child in timeseries_node['children']:
+            varname = timeseries_node_child['name'][len('states:'):]
```

Comment:
This code only adds states and time to the plots.
We should be plotting everything that's output in the timeseries.

This includes:
`time` - The time variable for the entire problem (generally starts at 0 at the beginning of the first phase)
`time_phase` - The elapsed time in the current phase
`states:<state_name}` - Integrated state variables
`controls:<control_name>` - Dynamic control variables
`control_rates:<control_name>_rate[2]` - Dynamic control rate (or second derivative if suffix is `rate2`.
`polynomial_controls:<control_name>` - Controls modeled by a polynomial across the entire phase
`polynomial_control_rates:<control_name>_rate[2] - Time derivatives of the polynomial controls
`parameters:<foo>` - Static parameters in the phase.
`<foo>` - Any ODE output that the user explicitly adds to the timeseries.